shared/js/background/broken-site-report.js

```diff
@@ -241,6 +252,50 @@ export async function breakageReportForTab ({
     if (errorDescriptions) brokenSiteParams.set('errorDescriptions', errorDescriptions)
     if (httpErrorCodes) brokenSiteParams.set('httpErrorCodes', httpErrorCodes)
     if (openerContext) brokenSiteParams.set('openerContext', openerContext)
+    if (reportFlow) brokenSiteParams.set('reportFlow', reportFlow)
+
+    return fire(pixelName, brokenSiteParams.toString())
+}
+
+/**
+ * Attempt to send a breakage report for the currently focused tab.
+ *
+ * @param {Object} arg
+ * @prop {string} pixelName
+ * @prop {import("./classes/tab") | undefined} arg.currentTab
+ * @prop {string | undefined} arg.category
+ * @prop {string | undefined} arg.description
+ * @prop {string | undefined} arg.reportFlow
+ *   String detailing the UI flow that this breakage report came from.
+ */
+export async function sendBreakageReportForCurrentTab (
```

Comment:
This function is only used in a component and a message handler - this seems a good candidate to be in a component itself so `settings` and `tdsStorage` can be injected. We can use the `registerMessageHandler` helper from the component to replace the statically defined `submitBrokenSiteReport` handler

Reply:
I had a go at moving the broken-site-report module into its own component today, but it's a fairly big change that I won't have time to tidy up and test properly for now. I don't think the changes here make refactoring that in the future any harder either, so I propose I file a follow-up task for this in Asana.